blockchain/My-Wallet-V3-iOS
# feat(Promises): Add currency pair model & promise based networking

## Objective

To simplify async network operations and give us a succinct way to deal with chaining API calls.
  
## Description

Using two dispatch queues, we pass the completion handler from the promised object down the chain so a CompleteFuture can have infinite .then methods chained. 

This is intended to work for the new exchange/KYC flow and be a scaffold for networking requests going forward.  Feedback is welcomed/encouraged while we consolidate and do away with the duplicitous networking stack and move towards a more universally extensible way to manage networking going forward with the goal of deprecating `NetworkManager` and `KYCNetworkRequest`.   

## How to Test

Open Settings and then tap `Identity Verification` It should start the KYC flow and should make the network call to the exchange. 

An example can be found in `KYCCordinator.swift`.  The main goal here is to start consolidation of network calls using a generic implementation of `URLSession` taking in a generic, decodable model.  Currently the network client supports `GET` requests.  

## Related Information

* Ticket Number:
IOS-1153

## Merge Checklist
